src/planner/match/MatchClausePlanner.cpp

```diff
@@ -225,10 +230,11 @@ Status MatchClausePlanner::rightExpandFromNode(const std::vector<NodeInfo>& node
     VLOG(1) << "root: " << subplan.root->outputVar() << " tail: " << subplan.tail->outputVar();
     auto left = subplan.root;
     NG_RETURN_IF_ERROR(
-        appendFetchVertexPlan(nodeInfos.back().filter,
+        MatchSolver::appendFetchVertexPlan(nodeInfos.back().filter,
                               matchClauseCtx->space,
                               matchClauseCtx->qctx,
                               edgeInfos.empty() ? initialExpr_->clone().release() : nullptr,
+                              "",
```

Comment:
I mean you should provide two overloaded functions with same name like following signature rather than updating these usage:

```cpp
static Status appendFetchVertexPlan(const Expression* nodeFilter, const SpaceInfo& space,  QueryContext* qctx, Expression* initialExpr, std::string inputVar, SubPlan& plan);

static Status appendFetchVertexPlan(const Expression* nodeFilter, const SpaceInfo& space,  QueryContext* qctx, Expression* initialExpr, SubPlan& plan);
```

in the second function, you could call the first overloaded one:
```cpp
Status appendFetchVertexPlan(const Expression* nodeFilter, const SpaceInfo& space,  QueryContext* qctx, Expression* initialExpr, SubPlan& plan) {
  return appendFetchVertexPlan(nodeFilter, space, qctx, initialExpr, plan.root->outputVar(), plan);
}
```

Reply:
Thanks for the illustration, I have updated this part